src/util/endianness_map.cpp

```diff
@@ -38,11 +38,12 @@ void endianness_mapt::build(const typet &src, bool little_endian)
 
 void endianness_mapt::build_little_endian(const typet &src)
 {
-  mp_integer s=pointer_offset_bits(src, ns); // error is -1
-  if(s<=0)
+  auto s = pointer_offset_bits(src, ns);
+
+  if(!s.has_value())
```

Comment:
` || !*s`

Reply:
I had pondered this one as well, but it's actually ok as the loop simply turns into a no-op.

Reply:
Ah true, though I suspect the original author intended to skip set-up for the loop in that case. That means the only proper error is the `!=` vs. `==` problem below.